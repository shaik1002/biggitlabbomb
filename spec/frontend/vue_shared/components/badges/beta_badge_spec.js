import { mount } from '@vue/test-utils';
import { GlBadge } from '@gitlab/ui';
import BetaBadge from '~/vue_shared/components/badges/beta_badge.vue';

describe('Beta badge component', () => {
  let wrapper;

  const findBadge = () => wrapper.findComponent(GlBadge);
  const createWrapper = (props = {}) => {
    wrapper = mount(BetaBadge, {
      propsData: { ...props },
    });
  };

  it('renders the badge', () => {
    createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('passes default size to badge', () => {
    createWrapper();

    expect(findBadge().props('size')).toBe('md');
  });

  it('passes given size to badge', () => {
    createWrapper({ size: 'sm' });

    expect(findBadge().props('size')).toBe('sm');
  });
});
