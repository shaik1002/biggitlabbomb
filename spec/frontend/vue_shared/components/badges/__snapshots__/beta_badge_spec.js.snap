// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beta badge component renders the badge 1`] = `
<div>
  <a
    class="badge badge-neutral badge-pill gl-badge gl-cursor-pointer md"
    href="#"
    target="_self"
  >
    Beta
  </a>
  <div
    class="gl-popover"
  >
    <p>
      A Beta feature is not production-ready, but is unlikely to change drastically before it's released. We encourage users to try Beta features and provide feedback.
    </p>
    <p
      class="gl-mb-0"
    >
      A Beta feature:
    </p>
    <ul
      class="gl-pl-4"
    >
      <li>
        May be unstable.
      </li>
      <li>
        Should not cause data loss.
      </li>
      <li>
        Is supported by a commercially reasonable effort.
      </li>
      <li>
        Is complete or near completion.
      </li>
    </ul>
  </div>
</div>
`;
